Add example genrules that break SourceKit LSP's indexing & provide fix

josh-arnold-1 opened the pull request:
This genrule actually breaks indexing! You can see SourceKit LSP tries to index the genrule itself...

[11:02:04 AM] 🟨🟫⬛️ Indexing /Users/.../sourcekit-bazel-bsp/Example/HelloWorld/GenerateDummySwiftFile
[11:02:04 AM] 🟨🟫⬛️ clang: error: no input files
[11:02:04 AM] 🟨🟫⬛️ Finished with exit code 1 in 0.00619425 seconds

I'm not sure how to fix this — any help would be appreciated, thanks!

josh-arnold-1 committed:
Add an example GenerateDummySwiftFile genrule

This genrule actually breaks indexing! You can see SourceKit LSP tries to index the genrule itself...

[11:02:04 AM] 🟨🟫⬛️ Indexing /Users/.../sourcekit-bazel-bsp/Example/HelloWorld/GenerateDummySwiftFile
[11:02:04 AM] 🟨🟫⬛️ clang: error: no input files
[11:02:04 AM] 🟨🟫⬛️ Finished with exit code 1 in 0.00619425 seconds

I'm not sure how to fix this — any help would be appreciated, thanks!

rockbruno committed:
Improve performance of parsing compiler arguments (#4)

* Allow indexstore arguments to be added in swift invocations

* Improve performance of processing compiler arguments

josh-arnold-1 committed:
Update HelloWorld to use 'expand_template_rule'

Add another example that causing this indexing error to occur

[11:16:39 AM] 🟪🟩⬛️ clang: error: no input files
[11:16:39 AM] 🟨⬜️🟨 clang: error: no input files
[11:16:39 AM] 🟪🟩⬛️ Finished with exit code 1 in 0.009771333 seconds
[11:16:39 AM] 🟨⬜️🟨 Finished with exit code 1 in 0.00974475 seconds

josh-arnold-1 committed:
Update HelloWorld to use 'expand_template_rule'

Add another example that causing this indexing error to occur

[11:16:39 AM] 🟪🟩⬛️ clang: error: no input files
[11:16:39 AM] 🟨⬜️🟨 clang: error: no input files
[11:16:39 AM] 🟪🟩⬛️ Finished with exit code 1 in 0.009771333 seconds
[11:16:39 AM] 🟨⬜️🟨 Finished with exit code 1 in 0.00974475 seconds

rockbruno committed:
README: Add more links, add basic info about rulesets

rockbruno committed:
Example README: Detail exactly when each tab shows up

rockbruno committed:
Example README: More detailed troubleshooting steps

josh-arnold-1 committed:
Provide a fix for the issues

It turns out we're still returning targets to index even when they have no files. This change should fix the issue.

josh-arnold-1 commented:
I found a fix, added it as the most recent commit

rockbruno committed:
Add CLAUDE.md

rockbruno committed:
Add info about our Slack channel

josh-arnold-1 commented on Sources/Requests/TextDocumentSourceKitOptionsHandler.swift:
```diff
@@ -53,6 +53,12 @@ final class TextDocumentSourceKitOptionsHandler {
             request.language,
             request.textDocument.uri,
         )
+
+        // If no compiler arguments are found, return nil to avoid sourcekit indexing with no input files
+        if args.isEmpty {
+            return nil
+        }
+
```
Oh, I didn't check that — I'm not sure if there is an obvious way to determine if a file is generated, unless I'm mistaken? What are your thoughts?

rockbruno replied:
In any case I think the nil change should be correct anyway, so I'll merge this one

josh-arnold-1 committed:
Revert "Improve performance of parsing compiler arguments" and add unit tests to prevent future regressions in aquery parsing logic (#8)

* Fix warning about unused variable

* Revert "Improve performance of parsing compiler arguments (#4)"

This reverts commit 3d22589d548973ae6e5b3eb26e15a5ae1f21e3c7.

* Add unit test for existing compiler arg parsing

- Extract the 'processCompilerArguments' function into its own file, so its easy to test

- Add 'ParseAqueryTests' which we can use to validate there are no regressions in future parsing logic changes

rockbruno committed:
Add support for long logs, log compiler args on debug (#6)

* Add support for long logs, log compiler args on debug

* Update Shell.swift

* Update Shell.swift

rockbruno committed:
Prepare the repo for CI and testing (#10)

* Separate the executable code from the library code

* Add base test target for the BSP

* Run swift-format

* Minor logger improvements

* Minor logger improvements (2)

* Move ParseAQueryTests to Swift Testing

* Lint SplitLogMessage

* Remove now unnecessary base test

rockbruno committed:
Clean-up the BSPServer entrypoint code (#11)

rockbruno committed:
Refactor PrepareHandler logic

rockbruno committed:
Refactor TargetSourcesHandler logic

BalestraPatrick committed:
Add GH actions configuration (#13)

rockbruno committed:
Separate the base BSP handling logic from this tool's implementation (#12)

* Move around BSP-related files, re-structure task logger

* Add a generalized BSP message handler

* Add tests for BSPMessageHandler

* Use dynamic registration for BSPMessageHandler

* Move all state to separate class, add it to the lock

rockbruno committed:
Add refactored handlers for initialize and the shutdown requests (#15)

rockbruno committed:
Add more build and test actions to the CI (#16)

* Add more build and test actions to the CI

* Make HelloWorldTests pass

rockbruno commented:
Probably conflicted because of the other revert PR, if you rebase it I can merge it in 👍

rockbruno committed:
Divide build target querying into multiple classes (#17)

rockbruno committed:
Add base re-structured SKOptionsHandler

rockbruno committed:
Add base re-structured WatchedFileChangeHandler

rockbruno committed:
Clean-up deprecated files, wire in new handlers

rockbruno committed:
Adjust lock handling, upgrade example project to release builds

rockbruno committed:
Separate logs per-file

rockbruno committed:
Use proper OSLog privacy levels

rockbruno committed:
Add instructions on how to see redacted logs and debug

sean7218 committed:
BazelProtobufBindings for Parsing ActionGraph (#14)

* Add Protobuf Definitions and Action Parser Tests

- Introduced `analysis_v2.proto` to define Protobuf messages for aquery data, including module information, rule definitions, and provider details.
- Created a README.md file for the Action Parser, outlining setup instructions and usage for generating Swift bindings from Protobuf definitions.
- Implemented unit tests in `ActionParserTests.swift` to validate the parsing of action graph data from Protobuf output, including tests for compiler arguments and action targets.
- Added sample Protobuf data file `actions.pb` for testing purposes.

* Slim down the binding files generated

  1. remove build.proto and build.pb.swift
  2. remove stardoc_output.proto and stardoc_output.pb.swift
  3. use Target defined in analysis_v2.proto instead of blaze_query.Target

* updated NOTICE for swift-protobuf

* rename ActionParser to BazelProtobufBindings
